<?xml version="1.0"?>
<launch>
    <arg name="roomba"/>
    <arg name="robot_frame_id" default="$(arg roomba)/base_link" />

    <group ns="$(arg roomba)" >
        <node pkg="pole_eliminator" type="pole_eliminator" name="pole_eliminator" >
            <param name="laser_frame_id" value="$(arg roomba)/laser" />
        </node>
        <node pkg="dwa_planner" type="dwa_planner" name="dwa_planner">
            <rosparam command="load" file="$(find navigator)/config/dwa_param.yaml"/>
            <param name="ROBOT_FRAME" value="$(arg robot_frame_id)" />
            <remap from="/local_map" to="local_map"/>
            <remap from="/scan" to="corrected_scan"/>
            <remap from="/local_goal" to="local_goal"/>
            <remap from="/cmd_vel" to="local_path/cmd_vel"/>
            <remap from="/odom" to="roomba/corrected_odometry"/>
        </node>
        <node pkg="trajectory_generator" type="trajectory_generator" name="trajectory_generator">
            <rosparam command="load" file="$(find trajectory_generator)/config/$(arg roomba)_param.yaml"/>
            <param name="ROOMBA" value="$(arg roomba)" />
            <param name="ROBOT_FRAME" value="$(arg robot_frame_id)" />
            <remap from="scan" to="corrected_scan"/>
            <remap from="pose" to="amcl_pose"/>
            <remap from="odom" to="roomba/corrected_odometry"/>
        </node>

        <node pkg="navigator" type="message_reviser_node" name="message_reviser" />
        <node pkg="navigator" type="waypoints_manager_node" name="waypoint_manager">
            <param name="RANDOM_WAYPOINTS" value="true" />
            <param name="GOAL_THRESHOLD" value="1.5" />
            <rosparam command="load" file="$(find navigator)/config/dkan_all_waypoints.yaml" />
            <rosparam command="load" file="$(find navigator)/config/dkan_half_points_relation.yaml" />
            <param name="FIRST_POSITION" value="5" />
            <param name="FIRST_LOCAL_GOAL" value="8" />
        </node>
    </group>
</launch>
